modin/developer/partition_unwrapper.py

```diff
@@ -0,0 +1,128 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+class PartitionUnwrapper(object):
+    """Class responsible for unwrapping axis partitions"""
+
+    @classmethod
+    def unwrap(cls, query_compiler, axis, engine=None, bind_ip=False):
+        """
+        Unwrap axis partitions of an API layer object using its `query_compiler`.
+
+        Parameters
+        ----------
+        query_compiler : BaseQueryCompiler
+            The query compiler of an API layer object.
+        axis : 0 or 1
+            The axis to unwrap partitions for (0 - row partitions, 1 - column partitions).
+        engine : str
+            The engine of the underlying partitions ("Ray", "Dask").
+        bind_ip : boolean, default False
+            Whether to bind node ip address to each axis partition or not.
+
+        Returns
+        -------
+        list
+            A list of Ray.ObjectRef to axis partitions of an `api_layer_object` if engine="Ray".
```

Comment:
```python
import modin.pandas as pd
import numpy as np
df = pd.DataFrame(np.arange(64))
from modin.developer import unwrap_row_partitions
# If `bind_ip=False`, then we get list of `Dask.Future` (in case we use Dask engine)
res = unwrap_row_partitions(df)
res
[<Future: finished, type: pandas.DataFrame, key: lambda-51300a5d-c6a8-4a84-98e7-f82e383d50b0>,
 <Future: finished, type: pandas.DataFrame, key: lambda-522443d4-616c-433a-9ad8-6078e57ed855>]
# if `bind_ip=True`, then we get list consisting of tuples of `Dask.Future` (in case we use Dask engine)
res = unwrap_row_partitions(df, bind_ip=True)
res
[(<Future: finished, type: builtins.str, key: lambda-f251e187-d27a-4053-abe3-2f19a9a33bec>,
  <Future: finished, type: pandas.DataFrame, key: lambda-7c2c61e6-ae0e-47a4-8e97-6577390eb9b3>),
 (<Future: finished, type: builtins.str, key: lambda-621b43b0-30f4-470e-89e6-e977099fabc6>,
  <Future: finished, type: pandas.DataFrame, key: lambda-6e860587-086d-40ce-a566-af39c198d46a>)]
```
The partitions are ordered, i.e.:
```python
# it's assumed we used `bind_ip=False
res[0].result()
     0
0    0
1    1
2    2
...
29  29
30  30
31  31
res[1].result()
     0
32  32
33  33
34  34
...
61  61
62  62
63  63
```
As for me, return values look good as it is now. Any concerns, @vnlitvinov , @devin-petersohn ?

Reply:
If order matters, I suggest rephrasing `maps A to B` into `pairs of A and B` (this comment was about docstring, not about the implementation)